woodwork/utils.py

```diff
@@ -298,3 +299,149 @@ def _parse_logical_type(logical_type, name):
+        validate_schema (bool, optional): Whether validation should be performed on the typing information
+            for the concatenated DataFrame. Defaults to True.
+
+    Returns:
+        object, type of objs: When objs contains at least one DataFrame, a Woodwork DataFrame is returned.
+            When concatenating along the columns (axis=1), a Woodwork DataFrame is returned.
+            When concatenating along the index (axis=1) with only series, a series is returned
+            without Woodwork initialized.
+    """
+    if not objs:
+        raise ValueError('No objects to concatenate')
+
+    # --> consider adding sort
+    # --> consider adding ignore index
+    table_name = ''
+    logical_types = {}
+    semantic_tags = {}
+    col_descriptions = {}
+    col_metadata = {}
+    table_metadata = {}
+    use_standard_tags = {}
+
+    index = None
+    time_index = None
+
+    concat_objs = []
+    # Record the typing information for all the columns that have Woodwork schemas
+    col_names_seen = set()
+    for obj in objs:
+        ww_columns = {}
+        obj_to_add = obj
+        if obj is None:
+            # --> not confident that this is the right choice. We need to be able to use the elements of objs to get the library type so it's hard if any element can be None
+            raise ValueError('Cannont include None in list of objects to concatenate')
+        elif isinstance(obj.ww.schema, ww.table_schema.TableSchema):
+            drop_cols = []
+            # Keep the first occurance of a key in metadata
+            table_metadata = {**obj.ww.metadata, **table_metadata}
+
+            # Combine table names
+            if obj.ww.name is not None:
+                if table_name:
+                    table_name += '_'
+                table_name += str(obj.ww.name)
+
+            # Handle table indexes
+            for idx_type, obj_idx, table_idx in [('index', obj.ww.index, index),
+                                                 ('time_index', obj.ww.time_index, time_index)]:
+                if obj_idx is not None:
+                    if table_idx is None:
+                        if idx_type == 'index':
+                            index = obj_idx
+                        elif idx_type == 'time_index':
+                            time_index = obj_idx
+                    elif obj_idx != table_idx:
+                        raise IndexError(f'Cannot concat dataframes with different Woodwork {idx_type} columns.')
+                    elif axis in {1, 'columns'} and obj_idx == table_idx:
+                        # Do not include the column from the dataframe to avoid duplicates columns
+                        drop_cols.append(table_idx)
+
+            # To avoid changing input objects, create a new DataFrame without the index columns
+            if drop_cols:
+                obj_to_add = obj.ww.drop(columns=drop_cols)
+            ww_columns = obj_to_add.ww.schema.columns
```

Comment:
Wanted to bring attention to this behavior here: In order to avoid having duplicate index and time index columns in the concatenated DataFrame, we need to only have these columns present once. We're keeping the first occurrence of them. 

It's not ideal to have to copy the data here, but the other option was the remove the column inplace with `pop`, but I didn't want to alter the input data. Happy to explore something different, if there's an option I missed.

Reply:
No longer relevant with behavior change to only allow one index and time index to be specified across the input objects.